imap_processing/spice/geometry.py

```diff
@@ -1 +1,128 @@
-"""Functions for computing geometry using SPICE."""
+"""Functions for computing geometry, many of which use SPICE."""
+
+from collections.abc import Iterable
+from enum import Enum
+from typing import NamedTuple, Optional, Union
+
+import numpy as np
+import spiceypy as spice
+
+from imap_processing.spice.kernels import ensure_spice
+
+
+class SpiceId(NamedTuple):
+    """Class that represents a unique identifier in the NAIF SPICE library."""
+
+    strid: str
+    numid: int
+
+
+class SpiceBody(Enum):
```

Comment:
This is great, I really like this style of defining these as Enums.

Do we ever run into a situation of possibly getting out of sync with the SPICE kernels? Should these be dynamically looked up from the Kernels themselves somehow?

Reply:
Yeah, good question. It would be great to parse certain kernels for this information. I'd have to think about how to approach that. For example, do we check those kernels into the repo?  The likelihood that any of the spacecraft or instrument values change should be small. I can write a ticket up for this but think that for now it is a nice to have that would take too much development time.

Reply:
:+1: Completely agree that this is very minor. I was just curious, so definitely punt on it into the very distant future.